prophet/src/app_data.rs

```diff
@@ -40,14 +37,16 @@ impl AppData {
 
     /// Clone the provided repositories and generate ReSSAs to analyze them
     /// based on the languages in its LAAST
-    pub fn from_repositories(mut repos: Repositories) -> Result<AppData, Error> {
+    pub fn from_repositories(mut repos: Repositories, ressa_dir: &str) -> Result<AppData, Error> {
```

Comment:
I think these references to ressa_dir could be changed to this at the top level
```rust
pub fn from_repositories<P: AsRef<Path>>(mut repos: Repositories, ressa_dir: P) -> Result<AppData, Error> {
   // call ressa_dir.as_ref() to get the path
}
```

and &Path everywhere else

Reply:
I also think you need to update the compat signature for this and update it as well since I don't see that here